src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -322,6 +332,76 @@ private async Task IndexAddressesContinuouslyAsync()
             }
         }
 
+        private void RewindOutPointRepository(uint256 hashToReorg)
+        {
+            Block blockToReorg = this.consensusManager.GetBlockData(hashToReorg).Block;
+
+            // For efficiency, only do one lookup to the block store. So we accumulate all needed transactions first.
+            var previousTransactionsToRetrieve = new HashSet<uint256>();
+
+            foreach (Transaction tx in blockToReorg.Transactions)
```

Comment:
You can filter here already by non-coinbase transactions?

Reply:
Yes, since a coinbase has no previous transactions it's spending from, we must ignore them for the purposes of rewinding a block.